src/utils/common-utils.ts

```diff
@@ -11,184 +11,49 @@ export enum ResourceType {
  * https://developer.mozilla.org/en-US/docs/Web/API/PerformanceResourceTiming/initiatorType
  */
 export enum InitiatorType {
-    /**
-     * IMAGES
-     * PerformanceResourceTiming with initiatorType=Input must be an image
-     * Per MDN docs: "if the request was initiated by an <input> element of type image.""
-     */
+    // IMAGES
     IMG = 'img',
     IMAGE = 'image',
     INPUT = 'input',
 
-    /**
-     * DOCUMENTS
-     */
+    // DOCUMENTS
     IFRAME = 'iframe',
     FRAME = 'frame',
 
-    /**
-     * SCRIPTS
-     */
+    // SCRIPTS
     SCRIPT = 'script',
 
-    /**
-     * STYLESHEETS
-     */
-    CSS = 'css'
-}
-
-/**
- * A PerformanceEntry or RumEvent that is sourced from the PerformanceAPI
- */
-export interface HasLatency {
-    startTime: DOMHighResTimeStamp;
-    duration: DOMHighResTimeStamp;
+    // STYLESHEETS
+    CSS = 'css',
+    LINK = 'link' // typically for stylesheets, but also for other things such as icons and fonts
 }
 
 /**
  * Creates key to link a RumEvent to the PerformanceEntry that it is sourced from
  * e.g. performanceKey(ResourceEvent) === performanceKey(PerformanceResourceTiming).
- * There is some worry of collision when startTime or duration are zero, such as when
- * resources are cached. But timestamps have not been observed to be zero in these cases.
  */
-export const performanceKey = (details: HasLatency) =>
-    [details.startTime, details.duration].join('#');
-
-const extensions = [
-    {
-        name: ResourceType.STYLESHEET,
-        list: ['css', 'less']
-    },
-    {
-        name: ResourceType.DOCUMENT,
-        list: ['htm', 'html', 'ts', 'doc', 'docx', 'pdf', 'xls', 'xlsx']
-    },
-    {
-        name: ResourceType.SCRIPT,
-        list: ['js']
-    },
-    {
-        name: ResourceType.IMAGE,
-        list: [
-            'ai',
-            'bmp',
-            'gif',
-            'ico',
-            'jpeg',
-            'jpg',
-            'png',
-            'ps',
-            'psd',
-            'svg',
-            'tif',
-            'tiff'
-        ]
-    },
-    {
-        name: ResourceType.FONT,
-        list: ['fnt', 'fon', 'otf', 'ttf', 'woff']
-    }
-];
-
-export const shuffle = (a: any[]) => {
-    for (let i = a.length - 1; i > 0; i--) {
-        const j = Math.floor(Math.random() * (i + 1));
-        const v = a[i];
-        a[i] = a[j];
-        a[j] = v;
+export const performanceKey = (startTime: number, duration: number) =>
```

Comment:
We are not using `name` it can be personally identifying information and omitted from RUM events

Reply:
question: It seems like we do use `name` if `recordResourceUrl` is `true`. Are we expecting to not use the `name` field at all?

Reply:
The configuration works reliably as is, since both startTime and duration are [DOMHighResTimeStamps](https://developer.mozilla.org/en-US/docs/Web/API/DOMHighResTimeStamp) and highly unique. We could improve `performanceKey()` by using name if it exists, and if not then use the current setup.

Reply:
Ah, I didn't see that we were talking about it in the context of just `performanceKey()` and thought we were talking about using the `name` field overall. I agree that `startTime` and `duration` should be unique enough!